### ISSUE TITLE
Feature request: make the launcher used for remote CPU autodetection jobs configurable

### ISSUE THREAD

**Issue by casparvl** [reframe-hpc/reframe#2926]
I ran into an issue where my CPU autodetection wasn't working because a) the `mpirun` command was configured as launcher and b) there was no system `mpirun` (i.e. other tests work, because they load _some_ flavour of MPI module). Thus, the CPU autodetection job that ReFrame produces:

```
#!/bin/bash
#SBATCH --job-name="rfm-detect-job"
#SBATCH --ntasks=1
#SBATCH --output=rfm-detect-job.out
#SBATCH --error=rfm-detect-job.err

_onerror()
{
    exitcode=$?
    echo "-reframe: command \`$BASH_COMMAND' failed (exit code: $exitcode)"
    exit $exitcode
}

trap _onerror ERR

./bootstrap.sh
mpirun -np 1 ./bin/reframe --detect-host-topology=topo.json
```
Failed with an `mpirun: command not found` type of error. 

I could get it to work by configuring `srun` as the launcher for my partitions, but I _prefer_ to keep using `mpirun` as a launcher.

One solution would be if I could configure which launcher the CPU detection job uses, so that I could tell it to use the `local` launcher (there's no real reason for it to be launched with something like `srun` or `mpirun` anyway). I.e. something like setting an environment variable `RFM_REMOTE_DETECT_LAUNCHER`, and/or a config item in the reframe configuration file `reneral.remote_detect_launcher`. Would that be possible?

**Comment by casparvl** [comment]
An addition to this is that e.g. `prerun_cmds` set in the config are also not currently taken into account. I'm not sure if this is on purpose, but I can imagine scenarios where one might need these.

**Comment by vkarak** [comment]
@casparvl Can you give an example of this?

**Comment by casparvl** [comment]
Sorry, completely failed to notice you asked a follow-up question here. 

Well, the bootstrap _should_ work quite independently, so I guess there aren't many reasons to want to do `prerun_cmds`, but suppose there is no system python3 or pip, and I need to do something to make one available? I don't know, the `prerun_cmds` is just a very nice thing to be able to insert _something_ in that job script, for whatever reason one might have. Even just for debugging purposes in case people run into issues with the bootstrap or CPU autodetect, it's just easier to have a 'knob' that you can turn there to try different things (different python version, different pip, etc).

**Comment by vkarak** [comment]
@casparvl I agree with your original comment that the parallel launcher is not needed and indeed it has absolutely no meaning here. As for using the `prerun_cmds`, yes that could be a valid feature request, but I'd rather keep it as "nice to have" until somebody does hit this. I suggest therefore to close this issue and open two new ones:

1. One about removing the parallel launcher from the remote detection script (could be addressed in the next patch release)
2. One about supporting a `prerun_cmds`-like structure for the remote detection script.

**Comment by casparvl** [comment]
Created separate issue for (2) https://github.com/reframe-hpc/reframe/issues/2979

I think we can keep this issue for (1), since that was essentially the original topic of this issue, if that's ok with you?

**Comment by vkarak** [comment]
Closing this as (1) was done in #2978 and available in 4.3.3.
